static/js/controllers/configuration-translation-application.js

```diff
@@ -31,7 +34,9 @@ angular.module('inboxControllers').controller('ConfigurationTranslationApplicati
     };
 
     var updateTranslationModels = function() {
-      var lhsTranslation = findTranslation($scope.localeModel.lhs);
+      var lhsOption = $scope.localeModel.lhs === TRANSLATION_KEYS_OPTION.doc.code ?
+                        DEFAULT_LANGUAGE : $scope.localeModel.lhs;
```

Comment:
This is a little confusing to me because you're setting the language to default ('en') and then ignoring the translation in the template. This makes the code hard to read because it looks like the `lhs` in the model is going to be English but then the template ignores it and outputs the key anyway.

I think it'd be clearer to keep the template simple and add the logic in the controller, so around line 46 instead of having `lhs: lhs[key]` maybe have something like `lhs: showKeys ? key : lhs[key]` where `showKeys` is set to `$scope.localeModel.lhs === TRANSLATION_KEYS_OPTION.doc.code`. Then the template is really simple and the logic is testable and all in the same file.

What do you think?

Reply:
Sounds good. It does clean it. I am on it.